src/library.cpp

```diff
@@ -170,9 +171,11 @@ void Library::updateFromDir(QString monitorDir)
 
 void Library::asyncUpdateFromDir(QString dir)
 {
-    QtConcurrent::run( [=]() {
+    std::function f = [this, dir]() {
         updateFromDir(dir);
-    });
+    };
+    // wrap the lambda in LambdaRunnable for Qt 5.12 and later compat
+    QThreadPool::globalInstance()->start(new portutils::LambdaRunnable(f));
```

Comment:
I think that it would be better to provide a higher level utility `portutils::runConcurrently()` that would directly replace `QtConcurrent::run()`.

Reply:
`QtConcurrent::run()` returns a future and getting the return result blocks until the concurrent function has completed. Since this PR enforces checking return values, we can't use `QtConcurrent::run()` without blocking, or having some mechanism to come back and check later.

`QThreadPool::start()` is the [recommended approach](https://doc.qt.io/qt-6/qtconcurrentrun.html#returning-values-from-the-function) for fire-and-forget style requests.

Reply:
I think you misunderstood me. I meant that you could package your replacement code for `QtConcurrent::run()` (based on `QThreadPool::start()`) into a helper function `portutils::runConcurrently()` and then would only have replace the calls to  `QtConcurrent::run()` with calls to `portutils::runConcurrently()`. That would hide the implementation details and make the code look cleaner/more readable.

Reply:
On another note, since the purpose of this PR is to eliminate compilation warnings rather than improve portability, can't you suppress the warning in this case in a simpler way like

```c++
(void)QtConcurrent::run( [=]() { ... });
```

or

```c++
const auto unused = QtConcurrent::run( [=]() { ... });
Q_UNUSED(unused);
```

Reply:
Fair enough, its definitely easier to simply ignore return values with `(void)` and `QtConcurrent` is easier to use. I restored `QtConcurrent::run()` and just ignored the return value.